x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/monitor_details/monitor_status/use_monitor_status_data.ts

```diff
@@ -5,75 +5,132 @@
  * 2.0.
  */
 
-import { useCallback, useMemo, useState } from 'react';
-import { throttle } from 'lodash';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useDebounce } from 'react-use';
+import { useLocation } from 'react-router-dom';
 
-import { scheduleToMinutes } from '../../../../../../common/lib/schedule_to_time';
 import { useSyntheticsRefreshContext } from '../../../contexts/synthetics_refresh_context';
 
 import { useSelectedMonitor } from '../hooks/use_selected_monitor';
-import { usePingStatuses } from '../hooks/use_ping_statuses';
 import {
   dateToMilli,
   createTimeBuckets,
-  createStatusTimeBins,
   CHART_CELL_WIDTH,
   indexBinsByEndTime,
   MonitorStatusPanelProps,
+  createStatusTimeBins,
+  MonitorStatusTimeBin,
 } from './monitor_status_data';
+import { useSelectedLocation } from '../hooks/use_selected_location';
+import {
+  clearMonitorStatusHeatmapAction,
+  quietGetMonitorStatusHeatmapAction,
+  selectHeatmap,
+} from '../../../state/status_heatmap';
+
+type Props = Pick<MonitorStatusPanelProps, 'from' | 'to'> & { initSize?: number };
```

Comment:
This comes from `initialSizeRef.current?.clientWidth`, would it be better if this received the `ref` itself instead and called the `clientWidth` inside this hook?

Reply:
++ I am fine with this. 6c9ee1f633dca9d2e99ba7cba428127d95148e20